### PR TITLE
Dependencies update 10 sep 2024

### PR DESCRIPTION
**Pull request by ninovanhooff**

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
-mockkAndroid = "1.13.9"
+mockkAndroid = "1.13.12"
 jvmTarget = "21"
-kotlin = "2.0.10"
-ksp = "2.0.10-1.0.24"
-gradlePlugin = "8.5.2"
+kotlin = "2.0.20"
+# should match kotlin https://github.com/google/ksp/releases
+ksp = "2.0.20-1.0.25"
+gradlePlugin = "8.6.0"
```
Comment (ninovanhooff) [inline gradle/libs.versions.toml:11]: Gradle plugin version 8.6.0 seems to require java version 22, which does not match our target compatibility of 21. Also our self-hosted runner seems to be running java_22 (edit 2025-09-29: typo? is the runner using 21?) which is recommended by this template. So before merging, it seems we should either update the runner to support 22 and change the targetcompatibility to 22...

or revert this gradle version change

Comment (ninovanhooff) [reply gradle/libs.versions.toml:11]: https://www.notion.so/q42/GitHub-Actions-CI-op-onze-self-hosted-runner-Mac-Mini-7f7d2e6312444b98be521394223db6e2

https://docs.gradle.org/current/userguide/compatibility.html?_ga=2.163671295.1169800436.1726069472-1934319561.1716894485&_gl=1*1353ja0*_gcl_au*MTAxNzIyNTU2MS4xNzI1MjkyOTU5*_ga*MTkzNDMxOTU2MS4xNzE2ODk0NDg1*_ga_7W7NC6YNPT*MTcyNjA2OTQ3Mi41LjEuMTcyNjA2OTQ4Ni40Ni4wLjA.#android

Comment (ninovanhooff) [reply gradle/libs.versions.toml:11]: With all of the above, I hardly know what I'm talking about

Comment (ninovanhooff) [reply gradle/libs.versions.toml:11]: Update: even without this change; the self-hosted build fails, this change may not be the cause and could actually be fine https://github.com/Q42/Template.Android/issues/79

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
 googleServices = "4.4.2"
 crashlyticsPlugin = "3.0.2"
-firebaseBOM = "33.1.2"
-manesVersions = "0.44.0"
-littleRobotsCatalogUpdates = "0.8.1"
+firebaseBOM = "33.2.0"
```
Comment (izadiegizabal) [inline gradle/libs.versions.toml:14]: the `33.3.0` has already been released: https://firebase.google.com/support/release-notes/android#2024-09-12

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -1,30 +1,29 @@
...
 hilt = "2.52"
 retrofit = "2.11.0"
-kotlinx-serialization = "1.6.3"
+kotlinx-serialization = "1.7.2"
```
Comment (izadiegizabal) [inline gradle/libs.versions.toml:17]: `1.7.3` has been released: https://github.com/Kotlin/kotlinx.serialization/releases

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -1,30 +1,29 @@
...
 retrofit2KotlinxSerializationConverter = "1.0.0"
 networkResponseAdapter = "5.0.0"
 napier = "2.7.1"
 composeDestinations = "1.10.2"
 okhttp = "4.12.0"
-composePlatform = "2024.08.00"
-activityCompose = "1.9.1"
+composePlatform = "2024.09.00"
```
Comment (izadiegizabal) [inline gradle/libs.versions.toml:23]: compose bom `2024.09.02` has already been released: https://developer.android.com/develop/ui/compose/bom/bom-mapping

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -1,30 +1,29 @@
...
+activityCompose = "1.9.2"
 hiltNavigationCompose = "1.2.0"
-composeLifecycle = "2.8.4"
+composeLifecycle = "2.8.5"
```
Comment (izadiegizabal) [inline gradle/libs.versions.toml:26]: `2.8.6` has already been released: https://developer.android.com/jetpack/androidx/releases/lifecycle

**Review by izadiegizabal — COMMENTED** [review]
🆙

File: gradle/libs.versions.toml (hunk 1 above)
Comment (ninovanhooff) [reply gradle/libs.versions.toml:11]: self-hosted runners: mac-mini pcs somewhere @ Q office that runs builds. It is not enabled by default, that's why the builds are succeeding. See readme

**Review by izadiegizabal — APPROVED** [review]

File: gradle/libs.versions.toml (hunk 4 above)
Comment (ninovanhooff) [reply gradle/libs.versions.toml:23]: next time 👍

File: gradle/libs.versions.toml (hunk 5 above)
Comment (ninovanhooff) [reply gradle/libs.versions.toml:26]: next time 👍

**Pull request merged**